pkg/iac-providers/arm/v1/load-file.go

```diff
@@ -0,0 +1,279 @@
+	for _, config := range configs {
+		if _, present := a[config.Type]; !present {
+			a[config.Type] = []output.ResourceConfig{config}
+		} else {
+			resources := a[config.Type]
+			if !output.IsConfigPresent(resources, config) {
+				a[config.Type] = append(a[config.Type], config)
+			}
+		}
+	}
+}
+
+// getSourceRelativePath fetches the relative path of file being loaded
+func (a *ARMV1) getSourceRelativePath(sourceFile string) string {
+	// rootDir should be empty when file scan was initiated by user
+	if a.absRootDir == "" {
+		return filepath.Base(sourceFile)
+	}
+	relPath, err := filepath.Rel(a.absRootDir, sourceFile)
+	if err != nil {
+		zap.S().Debug("error while getting the relative path for", zap.String("IAC file", sourceFile), zap.Error(err))
+		return sourceFile
+	}
+	return relPath
+}
+
+func (a *ARMV1) getConfig(doc *utils.IacDocument, path string, m core.Mapper, r types.Resource,
+	vars map[string]interface{}) []output.ResourceConfig {
+
+	// handle linked template
+	if r.Type == deployments {
+		var parameters map[string]struct {
+			Value interface{} `json:"value"`
+		}
+		// get linked template
+		var templateData []byte
+		var templateSource string
+		if r.Properties != nil {
+			// nested template
+			if nestedTemplate, ok := r.Properties["template"]; ok {
+				templateData, _ = json.Marshal(nestedTemplate)
+				templateSource = a.getSourceRelativePath(path)
+			} else if templateLink, ok := r.Properties["templateLink"].(map[string]interface{}); ok {
+				// download from templatelink uri
+				if uri, ok := templateLink["uri"].(string); ok {
+					tempDir := filepath.Join(os.TempDir(), utils.GenRandomString(6))
+					defer os.RemoveAll(tempDir)
+					templatePath, err := a.downloadTemplate(uri, tempDir)
+					if err == nil {
+						templateData, _ = ioutil.ReadFile(templatePath)
+						templateSource = uri
+					}
+					// download from parametersLink uri
+					if parametersLink, ok := r.Properties["parametersLink"].(map[string]interface{}); ok {
+						if uri, ok := parametersLink["uri"].(string); ok {
+							parametersPath, err := a.downloadTemplate(uri, tempDir)
+							if err == nil {
+								parametersData, _ := ioutil.ReadFile(parametersPath)
+								err := json.Unmarshal(parametersData, &parameters)
+								if err != nil {
+									zap.S().Debug("unable to parse linked termplate parameters", zap.Error(err), zap.String("file", path))
+								}
+							}
+						}
+					}
+				} else if relativePath, ok := templateLink["relativePath"].(string); ok {
+					// relative path template
+					templatePath := filepath.Join(filepath.Dir(path), relativePath)
+					templateData, _ = ioutil.ReadFile(templatePath)
+					templateSource = a.getSourceRelativePath(templatePath)
+				}
+			}
+
+			// add parameters if specified on the deployment resource
+			if params, ok := r.Properties["parameters"]; ok {
+				paramString, _ := json.Marshal(params)
```

Comment:
should we handle the error case here too?

Reply:
Removed from this PR